src/include/optimizer/statistics/count_min_sketch.h

```diff
@@ -1,15 +1,23 @@
 #pragma once
 
 #include <algorithm>
+#include <cerrno>
 #include <cmath>
 #include <cstdlib>
 #include <cstring>
+#include <filesystem>
+#include <memory>
 #include <sstream>
+#include <string>
+#include <utility>
 #include <vector>
 
+#include "common/error/exception.h"
+#include "common/json.h"
 #include "common/macros.h"
 #include "loggers/optimizer_logger.h"
 #include "madoka/madoka.h"
+#include "storage/write_ahead_log/log_io.h"
```

Comment:
I think it might be better to move the Posix file I/O stuff out of the WAL directory and into it's own file I/O utility class. Thoughts?

Reply:
Yes, this sounds reasonable.